packages/http-client/__tests__/proxy.test.ts

```diff
@@ -145,6 +145,32 @@ describe('proxy', () => {
     expect(bypass).toBeFalsy()
   })
 
+  it('checkBypass returns true if host with subdomain in no_proxy', () => {
+    process.env['no_proxy'] = 'myserver.com'
+    const bypass = pm.checkBypass(new URL('https://sub.myserver.com'))
+    expect(bypass).toBeTruthy()
+  })
+
+  it('checkBypass returns false if no_proxy is subdomain', () => {
+    process.env['no_proxy'] = 'myserver.com'
+    const bypass = pm.checkBypass(new URL('https://myserver.com.evil.org'))
+    expect(bypass).toBeFalsy()
+  })
+
+  // Do not strip leading dots as per https://github.com/actions/runner/blob/97195bad5870e2ad0915ebfef1616083aacf5818/docs/adrs/0263-proxy-support.md
+  it('checkBypass returns false if host with leading dot in no_proxy', () => {
+    process.env['no_proxy'] = '.myserver.com'
+    const bypass = pm.checkBypass(new URL('https://myserver.com'))
+    expect(bypass).toBeFalsy()
+  })
+
+  // Do not match wildcard ("*") as per https://github.com/actions/runner/blob/97195bad5870e2ad0915ebfef1616083aacf5818/docs/adrs/0263-proxy-support.md
+  it('checkBypass returns true if no_proxy is "*"', () => {
+    process.env['no_proxy'] = '*'
+    const bypass = pm.checkBypass(new URL('https://anything.whatsoever.com'))
+    expect(bypass).toBeFalsy()
+  })
+
```

Comment:
Can We add one more test to chech a subdomain is supported when no_proxy has a leading dot?

```
  it('checkBypass returns true if host with subdomain in no_proxy defined with leading "."', () => {
    process.env['no_proxy'] = '.myserver.com'
    const bypass = pm.checkBypass(new URL('https://sub.myserver.com'))
    expect(bypass).toBeTruthy()
  })
```

Reply:
Ye sure! 

Another question also, since you want the `no_proxy` to be the same as in the [`actions/runner`](https://github.com/actions/runner), do you also want me to re-structure the code to be more similar to how they handle it? 
Obviously it's not the same language, but might be easier to keep consistency between them if they are implemented using the same logic, just with different languages.  

Might be overkill, but I'm willing to do it if that makes it easier maintainable for you guys!

Reply:
I'd prefer to keep PRs for the functional changes and refactoring changes separate, but really appreciate you asking 👍

Also, [here's](https://github.com/actions/runner/blob/5421fe3f7107f770c904ed4c7e506ae7a5cde2c2/src/Test/L0/RunnerWebProxyL0.cs#L342) a good set of unit tests from the runner code if needed

Reply:
I''ve added the [test you wrote](https://github.com/actions/toolkit/pull/1223#discussion_r1102785194), as well a new one for checking partial domains (such that `no_proxy=myserver.com` won't let `evilmyserver.com` through)